src/selfplay/game.cc

```diff
@@ -243,6 +245,18 @@ void SelfPlayGame::Play(int white_threads, int black_threads, bool training,
     }
 
     if (training) {
+      bool best_is_proof = best_is_terminal;  // But check for better moves.
+      if (best_is_proof && best_eval.wl < 1) {
+        auto best =
+            (best_eval.wl == 0) ? GameResult::DRAW : GameResult::BLACK_WON;
+        auto upper = best;
+        for (const auto& edge : node->Edges()) {
+          upper = std::max(-edge.GetBounds().first, upper);
```

Comment:
I'm a bit confused about this negation.  bounds is an enum with range 1->3 for valid values.  Negating here means you are taking max between a valid gameresult and an invalid gameresult?